### PR TITLE
CITATION.cff: create citation file

### PR DESCRIPTION
**Pull request by forrestfwilliams**
This pull request creates a `CITATION.cff` file in the root directory of the repository, which will allow GitHub to create a "CITE AS" section on the readme page.

### CODE REVIEW

**Comment by asjohnston-asf** [comment]
.cff file reference: https://citation-file-format.github.io/

**Comment by forrestfwilliams** [comment]
Right now this reference points to the main HyP3 directory's DOI, but I'm unsure if this is the behavior we want.

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  given-names: "Forrest"
+  orcid: "https://orcid.org/0000-0001-8721-6020"
+- family-names: "Herrmann"
+  given-names: "Jake"
+- family-names: "Smale"
+  given-names: "Jacquelyn"
+date-released: 2020-07-10
+repository-code: "https://github.com/ASFHyP3/hyp3"
+title: "Hybrid Pluggable Processing Pipeline (HyP3): A cloud-based infrastructure for generic processing of SAR data"
+doi: 10.5281/zenodo.6568160
```
Comment (jhkennedy) [inline CITATION.cff:31]:
```suggestion
doi: 10.5281/zenodo.4646138
```
I think we want to point to the hyp3-docs DOI for this DOI. It's common to cite the documentation as the citation for the overall concept and the specific repos/releases as well.

**Review by jhkennedy — CHANGES_REQUESTED** [review]
We should also add a `.zenodo` file so the zenodo metadata maches this -- that could be in this PR or another (I lean towards the former)

Likewise, we'll want to do the same for at least hyp3, the plugins, etc. But probably all of our repositories.

**Comment by forrestfwilliams** [comment]
> We should also add a `.zenodo` file so the zenodo metadata maches this -- that could be in this PR or another (I lean towards the former)
> 
> Likewise, we'll want to do the same for at least hyp3, the plugins, etc. But probably all of our repositories.

We've decided to move this addition to a future PR.

**Comment by asjohnston-asf** [comment]
LGTM, here's how it renders:
`Hogenson, K., Kristenson, H., Kennedy, J., Johnston, A., Rine, J., Logan, T., Zhu, J., Williams, F., Herrmann, J., Smale, J., & Meyer, F. (2020). Hybrid Pluggable Processing Pipeline (HyP3): A cloud-native infrastructure for generic processing of SAR data [Computer software]. https://doi.org/10.5281/zenodo.4646138`

**Review by asjohnston-asf — APPROVED** [review]

**Pull request merged**